src/control/lib/support/log.go

```diff
@@ -592,33 +756,45 @@ func collectServerLog(log logging.Logger, opts ...CollectLogsParams) error {
 	serverConfig.SetPath(cfgPath)
 	serverConfig.Load()
 
-	targetServerLogs, err := createHostLogFolder(serverLogs, log, opts...)
-	if err != nil {
-		return err
-	}
-
 	switch opts[0].LogCmd {
 	case "EngineLog":
 		if len(serverConfig.Engines) == 0 {
 			return errors.New("Engine count is 0 from server config")
 		}
 
+		targetServerLogs, err := createHostLogFolder(engineLogs, log, opts...)
+		if err != nil {
+			return err
+		}
+
 		for i := range serverConfig.Engines {
 			matches, _ := filepath.Glob(serverConfig.Engines[i].LogFile + "*")
-			for _, logfile := range matches {
-				err = cpLogFile(logfile, targetServerLogs, log)
-				if err != nil {
+			for _, logFile := range matches {
+				err = cpLinesFromLog(log, logFile, targetServerLogs, opts...)
+				if err != nil && opts[0].Stop {
 					return err
 				}
 			}
 		}
+
 	case "ControlLog":
-		err = cpLogFile(serverConfig.ControlLogFile, targetServerLogs, log)
+		targetControlLogs, err := createHostLogFolder(controlLogs, log, opts...)
+		if err != nil {
+			return err
+		}
+
+		err = cpLogFile(serverConfig.ControlLogFile, targetControlLogs, log)
```

Comment:
Do the start and end date/time only apply to engines?

Reply:
Engine is the one which will have a lot of lines. Admin and control logs are not going to be big like engine so copying the full logs for control/Admin.

Reply:
Seems reasonable to me. Let's make sure it's clear in the documentation.

Reply:
ok I will extend the logic for all the other logs.